### PR TITLE
LB editor theming updates and header lock.

### LINKED ISSUE #274: Fix heading alignment on CTA component when aligned to the left
# Problem

Screen Shot 2020-07-08 at 4 24 34 PM

# Proposed Solution

Updated alignment on heading when left alignment is set within the component.

### LINKED ISSUE #1523: Lock header in LB for non-admin users
Possibly using this module: https://www.drupal.org/project/layout_builder_lock

### LINKED ISSUE #1908: Fix page title / featured image preview within layout builder edit screen
# Problem

There are a few issues that need to be addressed:

- The layout builder edit screen for pages contains a page title / hero section that is vertically condensed due to the `vh` property set on the hero container. 
- There are also multiple "Add Banner" options within the page title hero section - do we need to remove the ability to add blocks within this section?
- There is also no page title/breadcrumbs preview set if the page doesn't use a featured image. 

Screenshots:

Screen Shot 2020-07-07 at 10 59 30 AM
Screen Shot 2020-07-07 at 10 58 55 AM

# Proposed Solution(s)

1. Adjust styles for hero component within `admin.scss` in uids_base 
2. Investigate method to show page title block by default and only show page title/hero component if a feature image is set. 
3. Remove option to add custom blocks within the Header Layout with Background Media

### LINKED ISSUE #2029: Refine Layout Builder "Preview"
Per some of the feedback from the retro we had after our research spike into Layout Builder, we should refine the way the preview looks when people are actually using the Layout Builder UI.

Some suggested steps:
- [x] Reduce or completely remove the padding inside regions and sections.
- [x] Remove or reduce the margin in between sections.
- [x] Make 'Add section' button and 'Add block' buttons look more similar, possibly add a background color.
- [x] Make 'Edge-to-edge' sections display that way in preview.
- [x] Remove dotted/dashed lines from around sections, regions, and blocks except when they are hovered over.

### LINKED ISSUE #2046: CTA block layout builder override class is being passed to headline via attributes
# Problem

Duplicate background patterns are appearing on uiowa.edu.

Screen Shot 2020-08-17 at 3 48 28 PM

# Proposed Solution:

Until this block is refactored, add `'attributes': attributes|without('class'),` to https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/templates/block/block--inline-block--uiowa-cta.html.twig#L45.

### PR DESCRIPTION
**Pull request by cory-skeers**
Resolves #2029, resolves #1523, resolves #1908, resolves #2046, resolves https://github.com/uiowa/uids/issues/274

Work toward providing a closer match between the LB 'preview' available in the editor and the final produced page.

### Testing
- `composer install`
- Sync a site.
- Check that the issues described in #2029 are resolved.
- Add a local webmaster and masquerade as them. Check that the header section is locked while masquerading but not as an admin.

### CODE REVIEW

**Comment by cory-skeers** [comment]
Did some initial work on this. Would love some extra eyes, either in the form of additional changes or feedback/suggestions.

**Comment by mark-bennett-uiowa** [comment]
Removed as this was a user error on my part, not an issue with the PR.

**Comment by mark-bennett-uiowa** [comment]
@cory-skeers is there a specific block(s) or section(s) I can test out? I was comparing this against a "live" prod site and wasnt seeing any differences with the naked eye.

**Comment by cory-skeers** [comment]
@mark-bennett-uiowa It looks like the updates aren't in place in those screenshots--did you `blt frontend` after pulling it? Should look like:

image

For the .layout-builder-block padding, the padding gets added only for the editor; removing it is part of trying to get the editor to look more closely like the final page (e.g. breadcrumbs sitting closer to the page title)

**Comment by mark-bennett-uiowa** [comment]
Changes look okay and no errors on my end. I think a little space around the Add Section button might be better, but the other changes look really good. Cleaner for sure.

**Comment by richardbporter** [comment]
This looks good, thanks. I'm going to close #2039 and make part of this. It would be nice to address https://github.com/uiowa/uiowa/issues/1908 here as well.

File: docroot/themes/custom/uids_base/scss/admin.scss [hunk 1]
```diff
@@ -66,17 +73,22 @@
...
-  outline: 2px dotted #e6ecf8;
+  margin-bottom: 0;
+
+  &:hover {
+    outline: 2px dotted #0036b1;
+    outline-offset: 8px;
+  }
+
+  .hero {
+    height: 30vh;
```
Comment (richardbporter) [inline docroot/themes/custom/uids_base/scss/admin.scss]: I'm not sure if this is the best way to address part of #1908.

**Comment by richardbporter** [comment]
The patch in a475c7b, in some form, should address the adding blocks part of #1908.

File: docroot/themes/custom/uids_base/scss/admin.scss (hunk 1 above)
Comment (bspeare) [reply docroot/themes/custom/uids_base/scss/admin.scss]: I think we want to do something like `height: auto;` for now until the Hero refactor is complete.  The `vh` has added specifically for the home page video to stay within the viewport.

Comment (richardbporter) [reply docroot/themes/custom/uids_base/scss/admin.scss]: Yes that works much better. 👍

**Comment by bspeare** [comment]
I'm going to test this a little more this afternoon.

**Comment by richardbporter** [comment]
Sounds like I need to switch to a D7 bug that came up. Feel free to jump in here.

**Comment by bspeare** [comment]
It looks like the "Hero" still appears in the layout builder edit screen because there is placeholder content set for the Featured Image.    Do we need placeholder content for the Featured Image?  Where is that set?

Screen Shot 2020-08-14 at 2 20 04 PM

The template is checking here: https://github.com/uiowa/uiowa/blob/lb-editor-theming/docroot/themes/custom/uids_base/templates/layouts/onecol--background/layout--onecol--background.html.twig#L23 for the background image and not displaying the normal page title because of the placeholder content.

File: docroot/themes/custom/uids_base/templates/layouts/onecol--background/layout--onecol--background.html.twig [hunk 2]
```diff
@@ -21,16 +21,15 @@
...
 
 {% if background %}
-
   {% embed "@uids_base/uids/hero.html.twig" with {
     hero_alignment : 'bottom',
     'attributes' : attributes.addClass(classes)
   } %}
     {% block hero_image %}
-      {{ content.background }}
+      {{ content.background | without('layout_builder_add_block')}}
```
Comment (richardbporter) [inline docroot/themes/custom/uids_base/templates/layouts/onecol--background/layout--onecol--background.html.twig]: I think these can be removed now. The buttons should not show up with the patch applied to lb_direct_add.

**Comment by richardbporter** [comment]
> It looks like the "Hero" still appears in the layout builder edit screen because there is placeholder content set for the Featured Image. Do we need placeholder content for the Featured Image? Where is that set?
> 
> The template is checking here: https://github.com/uiowa/uiowa/blob/lb-editor-theming/docroot/themes/custom/uids_base/templates/layouts/onecol--background/layout--onecol--background.html.twig#L23 for the background image and not displaying the normal page title because of the placeholder content.

I gave this a shot using similar code to what we have in other preprocess functions. The hero component is not embedded now if the featured image field is not set.

**Comment by richardbporter** [comment]
I think the only thing left here is solving for https://github.com/uiowa/uiowa/issues/2029#issuecomment-674994969.

**Comment by bspeare** [comment]
**Some comments from an editor working on coronavirus.uiowa.edu site:**

Confusion over "Configure Section" link vs the "Edit Block".    Since "Configure Section" is the most prominent link visually, the expectation is that they will be able to edit the blocks by clicking that link. 
  - Can we enhance the visibility of the "Edit Block"?

Confusion over what each block will look like.
 -  Could we add small icons next to the blocks on the "Add Block" so that the page builder can get a sense of what they are using?

**Comment by bspeare** [comment]
https://www.drupal.org/project/lb_ux (https://stm5f3bf4a639f28-bddtnlvocieisvf9hpnnxrgu4yaprjzw.tugboat.qa/) does a nice job on the "Configure Section" and "Edit Block" by only showing these options when the user hovers over them.    By doing that it eliminates the visual issue with "Configure Section" always being front and center. 
# Section
Screen Shot 2020-08-18 at 10 39 14 AM

# Block 
Screen Shot 2020-08-18 at 10 39 19 AM

**Comment by bspeare** [comment]
> I think the only thing left here is solving for [#2029 (comment)](https://github.com/uiowa/uiowa/issues/2029#issuecomment-674994969).

@richardbporter I'll take another look at this now that my VM is working.

**Comment by cory-skeers** [comment]
@bspeare lb_ux looks pretty slick. Tried putting it on locally, and there'd need to be some work put in to get it to play nicely with some of the other updates, but I think either that or making our own version of just those parts of it would be a good investment.

**Comment by bspeare** [comment]
@cory-skeers I'm adding a quick adjustment now that I'll push up shortly that might help.

**Comment by bspeare** [comment]
I just added an update that hides the "Configure Section" link by default and will show it on focus or hover.    Thoughts?

Screen Shot 2020-08-18 at 1 03 47 PM

**Comment by bspeare** [comment]
@richardbporter I tested https://github.com/uiowa/uiowa/issues/2029#issuecomment-674994969 and it appears styled:

Screen Shot 2020-08-18 at 2 42 32 PM

Did you have this issue with the About page on the default site?

**Comment by richardbporter** [comment]
> Did you have this issue with the About page on the default site?

Yes. Maybe there is something wrong/missing with my local environment?

**Comment by richardbporter** [comment]
New styles look good. 👍

**Comment by bspeare** [comment]
@richardbporter Maybe I set up the local default site wrong.  Can we can another person to test this?

**Comment by cory-skeers** [comment]
This is looking really good!

@bspeare @richardbporter I'm seeing the sidebar as styled (and not stacked like in https://github.com/uiowa/uiowa/issues/2029#issuecomment-674994969) in my local dev.

**Comment by pyrello** [comment]
Drat. I wasn't paying attention to my email and missed that you were pushing updates to this. I was working on some separate updates and I was also cleaning up the SCSS as I was going so the merge is going to be messy. I guess I'll revisit tomorrow.

**Comment by pyrello** [comment]
Since it looks like @bspeare went in a different direction with this, I'm not sure that it is worth it to try to resolve the difference between what I worked on and what he created. Here are some screen shots and you can tell me if you think it is worth it to try to sort it out.

No interaction:
Screen Shot 2020-08-18 at 6 46 24 PM

Hovering over 'Add section':
Screen Shot 2020-08-18 at 6 47 33 PM

Hovering over section:
Screen Shot 2020-08-18 at 6 47 53 PM

**Comment by richardbporter** [comment]
Really digging the grey colors instead of blue. 

If I was nitpicking... :)

I noticed when hovering a section, all the borders kinda make the page jump a bit:

image

I'd be interested in seeing the links (Add Section, Add Block, etc.) without the underline.

**Comment by richardbporter** [comment]
**Update**: I don't see an issue for this so I'm putting it here. I noticed that the CTA **Brain Pattern Black background** doesn't appear in the LB editor. 

image

**Comment by richardbporter** [comment]
Removing the transparent BG on LB blocks and the black BG set for brain black seems to fix it. Is the transparent background needed?

```diff --git a/docroot/themes/custom/uids_base/scss/admin.scss b/docroot/themes/custom/uids_base/scss/admin.scss
index a6546f9bb4..bc5c4e96f4 100644
--- a/docroot/themes/custom/uids_base/scss/admin.scss
+++ b/docroot/themes/custom/uids_base/scss/admin.scss
@@ -257,16 +257,13 @@
 
 
 .layout-builder-block {
-  //padding: 0;
   cursor: move;
-  background: transparent;
 
   &.bg--gray {
     background: $light;
   }
 
-  &.bg--black,
-  &.bg-pattern--brain-black {
+  &.bg--black {
     background: $secondary;
   }
```

**Comment by bspeare** [comment]
@richardbporter I added some background patterns back into this file, because I noticed this was occurring:

Screen Shot 2020-08-19 at 3 50 27 PM

**Comment by richardbporter** [comment]
> @richardbporter I added some background patterns back into this file, because I noticed this was occurring:
> 
> Screen Shot 2020-08-19 at 3 50 27 PM

Thanks. Sorry I didn't catch that.

**Comment by richardbporter** [comment]
Anything else we want to address here? Also, do we want to block this until an outcome for the lb_direct_patch is determined? https://www.drupal.org/project/lb_direct_add/issues/3165272

**Review by richardbporter — COMMENTED** [review]
Does a15842d resolve https://github.com/uiowa/uiowa/issues/2046?

**Comment by pyrello** [comment]
@richardbporter Yes

**Comment by pyrello** [comment]
Hello all. I think I am done with my adjustments to the preview screen. If you wouldn't mind taking another look and providing feedback, I would appreciate it.

The 'Add section' and 'Add block' have a light outline all the time to help indicate the space that they will fill and that it doesn't exist yet. When hovering over a section, light outlines appear around all the blocks to help the user see what they can interact with. When the user hovers over a block, the outline darkens.

**Comment by richardbporter** [comment]
Cool. I'm ok with this going in now. Anyone have other WIP on this PR?

**Review by richardbporter — APPROVED** [review]

**Comment by bspeare** [comment]
The block contextual links are now hidden by default and will appear when you hover over each block.  I'm done now.

**Review by richardbporter — APPROVED** [review]
Nice! I thought of that but forgot to mention it here.

**Pull request merged**